spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriber.java

```diff
@@ -1,156 +0,0 @@
-/*
- * Copyright 2013-2018 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.springframework.cloud.sleuth.instrument.reactor;
-
-import brave.Span;
-import brave.Tracer;
-import brave.Tracing;
-import brave.propagation.TraceContextOrSamplingFlags;
-import org.reactivestreams.Subscriber;
-import org.reactivestreams.Subscription;
-import reactor.util.Logger;
-import reactor.util.Loggers;
-import reactor.util.context.Context;
-
-import java.util.concurrent.atomic.AtomicBoolean;
-
-/**
- * A trace representation of the {@link Subscriber}
- *
- * @author Stephane Maldini
- * @author Marcin Grzejszczak
- * @since 2.0.0
- */
-final class SpanSubscriber<T> extends AtomicBoolean implements SpanSubscription<T> {
```

Comment:
Why have you removed this?

Reply:
SpanSubscriber was used only for lastOperation hook, which is not used anymore. Do you think it may be used outside sleuth? Should I revert and deprecate it?

Reply:
Ah ok. The class is indeed package private, but the static method calling it is public, isn't it? I would deprecate both the class and the static method and add a warning log statement that this method will be deleted in the next major release.